runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts

```diff
@@ -77,7 +78,12 @@ export default class ExcelActionHandler extends StatefulActionHandler {
     async getCell(
         input: ExcelGetCellActionInput
     ): Promise<string | number> {
-        return this.session.ActiveSheet.Range(`${input.column}${input.row}`).Value();
+        const optionalWorksheet = input?.worksheet;
+        const openedWorksheet = this.session.ActiveSheet.name;
+        if (optionalWorksheet) this.session.Worksheets(optionalWorksheet).Activate();
+        const cellValue = this.session.ActiveSheet.Range(`${input.column}${input.row}`).Value();
+        if (optionalWorksheet) this.session.Worksheets(openedWorksheet).Activate();
+        return cellValue;
```

Comment:
can't it be simplified like: this.session.Worksheets(optionalWorksheet).Range(`${input.column}${input.row}`).Value(), if optional worksheet exists?

Reply:
good point

Reply:
it does work!
done